### PR TITLE
[Quant][PT2E] Enable linear and linear-unary post-op gelu quant recipe for x86 inductor quantizer

### PR DESCRIPTION
**Pull request by Le-Zheng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116004
* #114854
* __->__ #114853

**Summary**
Add Gelu for linear-unary post-op quantization recipe to x86 inductor quantizer. 

**Test plan**
python -m pytest test/quantization/pt2e/test_x86inductor_quantizer.py -k test_linear_unary_gelu
python test/test_quantization.py -k test_linear_unary_with_quantizer_api

### CODE REVIEW

File: torch/ao/quantization/pt2e/prepare.py [hunk 1]
```diff
@@ -365,11 +365,13 @@ def _maybe_insert_input_observers_for_node(
         )
         new_args.append(new_arg)
 
-    # Clone has a memory_format kwarg and zeros_like has a pin_memory kwarg
-    # that persist in exported graph. This is just a work around for these.
+    # Clone has a memory_format kwarg, zeros_like has a pin_memory kwarg, and
+    # gelu has a memory kwarg that persist in exported graph.
```
Comment (leslie-fang-intel) [inline torch/ao/quantization/pt2e/prepare.py:369]: nit: the kwarg of gelu is not `memory` right? should be something relates with `approximate`?

**Review by jgong5 — APPROVED** [review]

File: torch/ao/quantization/pt2e/prepare.py (hunk 1 above)
Comment (Le-Zheng) [reply torch/ao/quantization/pt2e/prepare.py:369]: @leslie-fang-intel  Thank you for pointing out! Can I edit the comment as "# Clone has a memory_format kwarg, zeros_like has a pin_memory kwarg and gelu has kwarg of approximate  that persist in exported graph. This is just a work around for these." ?

**Review by leslie-fang-intel — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
@jerryzh168 Could you help to take a look of this PR? @Le-Zheng makes some change of [torch/ao/quantization/pt2e/prepare.py](https://github.com/pytorch/pytorch/pull/114853/files#diff-976c3b0c6f85048d3db01a0c394ce8eb16e2f7541f0983d0f4ef549baa4be822). So, this PR may need your kind approve to get landed.

**Comment by Le-Zheng** [comment]
@jerryzh168 could you take some time to review the changes I've made in this pull request [torch/ao/quantization/pt2e/prepare.py](https://github.com/pytorch/pytorch/pull/114853/files#diff-976c3b0c6f85048d3db01a0c394ce8eb16e2f7541f0983d0f4ef549baa4be822)? Thank you in advance for your time and input!

**Comment by jerryzh168** [comment]
sorry just saw this, does gelu has additional kwargs?

**Review by jerryzh168 — APPROVED** [review]
LG, but maybe we want to fix gelu kwargs in export, cc @tugsbayasgalan @zhxchen17 ?

**Pull request closed without merging**